packages/orm/src/reactiveHints.ts

```diff
@@ -161,21 +172,18 @@ export function reverseReactiveHint<T extends Entity>(
           return field.components.flatMap((comp) => {
             return reverseReactiveHint(rootType, comp.otherMetadata().cstr, subHint, undefined, false).map(
               ({ entity, fields, path }) => {
-                return { entity, fields, path: [...path, comp.otherFieldName] };
+                const otherFieldName = maybeAddTypeFilterSuffix(meta, comp);
+                return { entity, fields, path: [...path, otherFieldName] };
               },
             );
           });
         }
         case "m2m": {
-          const otherField =
-            field.otherMetadata().allFields[field.otherFieldName] ??
-            fail(`No field ${field.otherMetadata().type}.${field.otherFieldName}`);
-          const otherFieldName =
-            otherField.kind === "poly" ? `${field.otherFieldName}@${meta.type}` : field.otherFieldName;
```

Comment:
I extract this to `maybeAddTypeFilterSuffix` so it could be used across the relation `case`s